client/Packages/com.beamable.server/Editor/Usam/Editor/CodeService.cs

```diff
@@ -103,119 +107,187 @@ public async Promise Init()
+				var popup = BeamablePopupWindow.ShowUtility(Constants.Migration.MIGRATION_POPUP_NAME, migrationVisualElement, null,
+				                                            new Vector2(800, 400),  (window) =>
+				                                            {
+					                                            // trigger after Unity domain reload
+					                                            window.Close();
+				                                            });
+				migrationVisualElement.OnCancelled += popup.Close;
+				migrationVisualElement.OnClosed += popup.Close;
 			}
+			
+			
 			LogVerbose("Completed");
 		}
 
-		public async Promise Migrate()
+		public async Promise Migrate(List<IDescriptor> allDescriptors)
 		{
-			var commonCsProj = await MigrateCommon();
-			foreach (var microserviceDir in Directory.EnumerateDirectories("Assets/Beamable/Microservices"))
+			AssetDatabase.DisallowAutoRefresh();
+
+			foreach (IDescriptor descriptor in allDescriptors)
 			{
-				var microserviceName = Path.GetFileName(microserviceDir);
-				var path = $"{StandaloneMicroservicesPath}{microserviceName}/";
-				if (!Directory.Exists(path))
+				if (descriptor.ServiceType == ServiceType.MicroService)
 				{
-					LogVerbose($"Migrating {microserviceName} start");
-					await CreateMicroService(microserviceName, null);
-					return;
+					await MigrateMicroservice((MicroserviceDescriptor)descriptor);
 				}
+				else
+				{
+					await MigrateStorage((StorageObjectDescriptor)descriptor);
+				}
+			}
+			
+			// REMOVE OLD STUFF
+			var microPath = "Assets/Beamable/Microservices";
+			var storagePath = "Assets/Beamable/StorageObjects";
+			if (Directory.Exists(microPath))
+			{
+				Directory.Delete(microPath, true);
+			}
+			
+			if (Directory.Exists(storagePath))
+			{
+				Directory.Delete(storagePath, true);
+			}
+			
+			AssetDatabase.AllowAutoRefresh();
+			await Init();
+		}
 
-				var signpost = _services.FirstOrDefault(s => s.name.Equals(microserviceName));
-				if (signpost == null) continue;
-				var command = $"add {signpost.CsprojFilePath} reference {commonCsProj}";
-
-				await _dotnetService.Run(command);
+		private async Promise MigrateStorage(StorageObjectDescriptor storageDescriptor)
+		{
+			var storageName = storageDescriptor.Name;
+			var path = $"{StandaloneMicroservicesPath}{storageName}/";
+			var storageModel = MicroservicesDataModel.Instance.Storages.FirstOrDefault(s => s.Name == storageName);
+			var deps = MicroservicesDataModel.Instance.Services
+			                                 .Where(model => model.Dependencies.Any(s => s.Name == storageName)).Select(model => ServiceDefinitions.FirstOrDefault(d => d.BeamoId == model.Name))
+			                                 .ToList();
+			Debug.Log(storageModel);
+			Debug.Log(string.Join(", ", deps));
+			if (!Directory.Exists(path))
+			{
+				Debug.Log(storageName);	
+				await CreateStorage(storageName, deps, shouldInitialize: false);

```

Comment:
we'd need to copy the src code included in the storage as well. Is that happening, and I'm not seeing it?

Reply:
good catch, it isn't. I'm going to make that now